pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/FileUploadService.java

```diff
@@ -285,6 +297,42 @@ public int removeUnusedUploads(Handle handle, Instant olderThanTimestamp, Intege
         return uploadIdsToDelete.size();
     }
 
+    public void sendNotifications(final Handle handle) {
+        StreamEx.of(handle.attach(FileUploadDao.class).findWithoutSentNotification())
+                .groupingBy(FileUpload::getStudyId)
+                .forEach((studyId, fileUploads) -> sendNotifications(handle, studyId, fileUploads));
+    }
+
+    private void sendNotifications(final Handle handle, final Long studyId, final List<FileUpload> fileUploads) {
+        final var study = handle.attach(JdbiUmbrellaStudy.class).findById(studyId);
+        if (StringUtils.isBlank(study.getNotificationEmail())) {
+            log.info("Study {} doesn't have an e-mail for notifications", study.getGuid());
+            return;
```

Comment:
Not sure if we should update "notified at" field of the file and forget about that to avoid the situation when at some point study is updated with e-mail and a lot of emails are sent.